src/devices/MotorHat/MotorHat.cs

```diff
@@ -34,15 +39,17 @@ public class MotorHat : IDisposable
         /// </summary>
         /// <param name="settings">The I2C settings of the MotorHat.</param>
         /// <param name="frequency">The frequency in Hz to set the PWM controller.</param>
+        /// <param name="pinProvider">The <see cref="IMotorPinProvider"/> that provides <see cref="MotorPins"/> for various hats.</param>
         /// <remarks>
         /// The default i2c address is 0x60, but the HAT can be configured in hardware to any address from 0x60 to 0x7f.
         /// The PWM hardware used by this HAT is a PCA9685. It has a total possible frequency range of 24 to 1526 Hz.
         /// Setting the frequency above or below this range will cause PWM hardware to be set at its maximum or minimum setting.
         /// </remarks>
-        public MotorHat(I2cConnectionSettings settings, double frequency = 1600)
+        public MotorHat(I2cConnectionSettings settings, double frequency = 1600, IMotorPinProvider pinProvider = default!)
```

Comment:
To prevent an (unneccessary) breaking change, prefer adding a new constructor instead of adding an argument (even if it has a default value, this is binary breaking).

Also, declare it nullable instead of using an ! to assign null.

Reply:
I've added a separate constructor but am intrigued what you mean by "binary breaking change".

Reply:
A "binary breaking change" is one that would break binary compatibility. That means that by just exchanging the library (without recompiling the client code) such a change would break and cause a runtime error. This is relevant for the case where the library is used indirectly, and an update happens trough an indirect package update.